tools/scripts/stm32.mk

```diff
@@ -1,10 +1,92 @@
-ifndef STM32CUBE
-$(error STM32CUBE not found in shell environment.$(ENDL))
+ifeq ($(OS),Windows_NT)
+$(error STM32 builds on Windows are not currently supported.)
+else
+IDE = stm32cubeide
+STM32CUBEIDE ?= $(wildcard /opt/stm32cubeide)
+ifeq ($(STM32CUBEIDE),)
+$(error $(ENDL)$(ENDL)STM32CUBEIDE not defined or not found at default path /opt/stm32cubeide\
+		$(ENDL)$(ENDL)\
+		Please run command "export STM32CUBEIDE=/path/to/your/stm32cubeide"$(ENDL)\
+		Ex: export STM32CUBEIDE=/opt/stm32cubeide$(ENDL)$(ENDL))
+endif # STM32CUBEIDE check
+
+MX = STM32CubeMX
+STM32CUBEMX ?= $(wildcard /opt/stm32cubemx)
+ifeq ($(STM32CUBEMX),)
+$(error $(ENDL)$(ENDL)STM32CUBEMX not defined or not found at default path /opt/stm32cubemx\
+		$(ENDL)$(ENDL)\
+		Please run command "export STM32CUBEMX=/path/to/your/stm32cubemx"$(ENDL)\
+		Ex: export STM32CUBEMX=/opt/stm32cubemx$(ENDL)$(ENDL))
+endif # STM32CUBEMX check
+endif # OS check 
+
+# Locate the compiler path under STM32CubeIDE plugins directory
+COMPILER_BIN = $(realpath $(dir $(call rwildcard, $(STM32CUBEIDE)/plugins, *arm-none-eabi-gcc)))
+
+# Locate openocd location under STM32CubeIDE plugins directory
+OPENOCD_SCRIPTS = $(realpath $(addsuffix ..,$(dir $(call rwildcard, $(STM32CUBEIDE)/plugins, *st_scripts/interface/stlink-dap.cfg))))
+OPENOCD_BIN = $(realpath $(dir $(call rwildcard, $(STM32CUBEIDE)/plugins, *bin/openocd)))
+
+PLATFORM_RELATIVE_PATH = $1
+PLATFORM_FULL_PATH = $1
+
+# stm32 specific implementation of this function, sources need to go under Core directory
+relative_to_project = $(addprefix $(PROJECT_BUILD)/Core/, $(call get_relative_path, $1))
```

Comment:
Nitpick: Wouldn't it work with BUILD_DIR instead of PROJECT_BUILD. This way relative_to_project won't be duplicated in each file

Reply:
good point, i will investigate, i don't like duplicating either